src/painting/painter.rs

```diff
@@ -454,6 +454,9 @@ impl Painter {
     }
 
     /// Prints an external message
+    ///
+    /// This function doesn't flush the buffer. So buffer should be flushed
+    /// afterwards perhaps by repainting the prompt via `repaint_buffer()`.
     pub fn print_external_message(
```

Comment:
This should only be called in the context you describe so let's hide it
```suggestion
    pub(crate) fn print_external_message(
```